### PR TITLE
refactor: cobra.manipulation.validate

### PR DESCRIPTION
**Pull request by synchon**
* [x] description of feature/fix
* [x] tests added/passed

This PR adds typing annotations in `cobra.manipulation.validate`. It also upgrades the code style to Python 3.6+ standard.

### CODE REVIEW

**Comment by synchon** [comment]
Should be good for review.

File: src/cobra/manipulation/validate.py [hunk 1]
```diff
@@ -12,23 +17,48 @@
...
-        if reaction.annotation.get("sbo") not in NOT_MASS_BALANCED_TERMS:
+        if reaction.annotation.get("sbo") not in _NOT_MASS_BALANCED_TERMS:
             balance = reaction.check_mass_balance()
             if balance:
                 unbalanced[reaction] = balance
     return unbalanced
 
 
-def check_metabolite_compartment_formula(model):
+def check_metabolite_compartment_formula(model: "Model") -> List[str]:
```
Comment (cdiener) [inline src/cobra/manipulation/validate.py:43]: This only checks the formula. No idea why the compartment is mentioned in the function name. But this should at least be corrected in the Dichtring.

Comment (synchon) [reply src/cobra/manipulation/validate.py:43]: Yes you are right. I will replace the `compartments` by `formulae`.

**Review by synchon — COMMENTED** [review]
Should be corrected with the latest commit.

**Review by cdiener — APPROVED** [review]
Awesome, thanks!

**Pull request merged**